src/components/Label/Label.tsx

```diff
@@ -82,69 +82,45 @@ class Label extends UIComponent<Extendable<LabelProps>, any> {
   renderComponent({ ElementType, classes, rest, variables, styles }) {
     const { children, content, icon, iconPosition, image, imagePosition } = this.props
 
-    const imageElement =
-      image &&
-      Image.create(image, {
-        defaultProps: {
-          styles: styles.image,
-          variables: variables.image,
-        },
-      })
-
-    const iconElement =
-      icon &&
-      Icon.create(icon, {
-        defaultProps: {
-          styles: styles.icon,
-          variables: variables.icon,
-        },
-        overrideProps: this.handleIconOverrides,
-      })
-
-    let start: React.ReactNode = null
-    let end: React.ReactNode = null
-
-    // Default positioning of the image and icon
-    if (image && imagePosition === 'start') {
-      start = imageElement
-    }
-    if (icon && iconPosition === 'end') {
-      end = iconElement
-    }
-
-    // Custom positioning of the icon and image
-    if (icon && iconPosition === 'start') {
-      if (image && imagePosition === 'start') {
-        start = (
-          <React.Fragment>
-            {imageElement}
-            {iconElement}
-          </React.Fragment>
-        )
-      } else {
-        start = iconElement
-      }
-    }
-    if (image && imagePosition === 'end') {
-      if (icon && iconPosition === 'end') {
-        end = (
-          <React.Fragment>
-            {iconElement}
-            {imageElement}
-          </React.Fragment>
-        )
-      } else {
-        end = imageElement
-      }
+    if (childrenExist(children)) {
+      return (
+        <ElementType {...rest} className={classes.root}>
+          {children}
+        </ElementType>
+      )
     }
 
+    const imageElement = Image.create(image, {
+      defaultProps: {
+        styles: styles.image,
+        variables: variables.image,
+      },
+    })
+    const iconElement = Icon.create(icon, {
+      defaultProps: {
+        styles: styles.icon,
+        variables: variables.icon,
+      },
+      overrideProps: this.handleIconOverrides,
+    })
     return (
       <ElementType {...rest} className={classes.root}>
-        {childrenExist(children) ? (
-          children
-        ) : (
-          <Layout main={content} start={start} end={end} gap={pxToRem(3)} />
-        )}
+        <Layout
+          start={
+            <>
+              {imagePosition === 'start' && imageElement}
```

Comment:
Remove reassignment of local variables. We can simply render elements inline in their target position.